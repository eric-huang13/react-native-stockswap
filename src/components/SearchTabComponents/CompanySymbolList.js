import React, {Component} from 'react';
import {
  View,
  TouchableOpacity,
  StyleSheet,
  SafeAreaView,
  Text,
  FlatList,
} from 'react-native';

import {connect} from 'react-redux';
import {moderateScale} from '../../util/responsiveFont';

export class CompanySymbolList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      initialIndex: 4,
      stockData: [],
    };
  }
  componentDidMount() {
    this.props.stockCategory == 'gainers'
      ? this.setState({
          stockData: this.props.marketGainersTest,
        })
      : this.props.stockCategory == 'losers'
      ? this.setState({
          stockData: this.props.gainers,
        })
      : this.props.stockCategory == 'hbv'
      ? this.setState({
          stockData: this.props.hbv,
        })
      : null;
  }

  render() {
    const getItemLayout = (data, index) => ({
      length: moderateScale(89),
      offset: moderateScale(89) * index,
      index,
    });

    const {gainers, stockCategory, marketGainersTest} = this.props;

    return (
      <SafeAreaView style={style.mainContainer}>
        <FlatList
          keyExtractor={(item, index) => index.toString()}
          data={this.state.stockData}
          style={style.scollContainer}
          horizontal
          alignItems="center"
          showsHorizontalScrollIndicator={false}
          initialScrollIndex={this.props.itemId - 1}
          getItemLayout={getItemLayout}
          renderItem={({item, index}) => (
            <TouchableOpacity
              key={item.id}
              onPress={() =>
                this.props.navigation.navigate({
                  name: 'CompanyInformation',
                  params: {item},
                })
              }>
              <View style={style.symbolBox}>
                <Text
                  style={
                    this.props.ticker === item.ticker
                      ? {...style.symbol, backgroundColor: '#8B64FF'}
                      : {...style.symbol}
                  }>
                  {item.ticker}
                </Text>
              </View>
            </TouchableOpacity>
          )}
        />
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    gainers: state.company.gainers,
    losers: state.company.losers,
    highestByVolume: state.company.highestByVolume,

    marketGainersTest: state.company.marketGainersTest,
  };
};

export default connect(mapStateToProps)(CompanySymbolList);

const style = StyleSheet.create({
  mainContainer: {
    paddingVertical: moderateScale(10),
    height: moderateScale(60),
  },
  symbolBox: {
    marginHorizontal: moderateScale(12),
    width: moderateScale(63),
    height: moderateScale(23),
    alignContent: 'center',
    justifyContent: 'center',
  },
  symbol: {
    textAlign: 'center',
    backgroundColor: '#3e475b',
    fontFamily: 'Montserrat-Medium',
    color: '#FFFFFF',
    paddingVertical: moderateScale(4),
  },
});
