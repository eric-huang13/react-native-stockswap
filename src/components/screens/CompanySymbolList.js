import React, {Component} from 'react';
import {
  View,
  TouchableOpacity,
  StyleSheet,
  ScrollView,
  SafeAreaView,
  Text
} from 'react-native';

import {connect} from 'react-redux';
import CompanyBox from './CompanyBox';

export class CompanySymbolList extends Component {
    constructor(props) {
        super(props);
   
      }
 

  render() {
    const {gainers} = this.props;
    console.log(this.props.navigation,'props in symbol list')
    return (
      <SafeAreaView style={style.mainContainer}>        

         
              <ScrollView
              style={style.scollContainer}
                horizontal={true}
                showsHorizontalScrollIndicator={false}
                alignItems='center'
                justifyContent="space-between">
                {gainers.map((item) => {
                  return (
                    <TouchableOpacity
                      key={item.id}
                      onPress={() =>
                        this.props.navigation.navigate({
                          name: 'CompanyInformation',
                          params: {item},
                        })
                      }>
                        <View style={style.symbolBox}>
                      <Text style={style.symbol}>{item.symbol}</Text>
                      </View>
                    </TouchableOpacity>
                  );
                })}
              </ScrollView>
    
      </SafeAreaView>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    gainers: state.company.gainers,
    losers: state.company.losers,
    highestByVolume: state.company.highestByVolume,
  };
};


export default connect(mapStateToProps)(CompanySymbolList);

const style = StyleSheet.create({
  mainContainer: {
    //   borderWidth:1,
      paddingVertical:10,
      height:60,
    
  },
  scollContainer:{
// borderWidth:1,

  },
  symbolBox:{
    // borderWidth:1,
    marginHorizontal:12,
    width:63,
    height:23,
    alignContent:'center',
    justifyContent:'center'
  },
  symbol:{   
    textAlign:'center',
    backgroundColor:'#8B64FF',
    fontFamily:'Montserrat-Medium',
    color:'#FFFFFF',
    paddingVertical:4,
  

  },

  
});
