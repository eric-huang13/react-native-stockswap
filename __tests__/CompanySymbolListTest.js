import React from 'react';
import CompanySymbolList from '../src/components/SearchTabComponents/CompanySymbolList';
import ShallowRenderer from 'react-test-renderer/shallow'; 


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<CompanySymbolList/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});