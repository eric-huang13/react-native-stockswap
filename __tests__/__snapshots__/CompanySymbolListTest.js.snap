// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "height": 84.74452554744525,
      "paddingVertical": 14.124087591240876,
    }
  }
>
  <FlatList
    alignItems="center"
    disableVirtualization={false}
    getItemLayout={[Function]}
    horizontal={true}
    initialNumToRender={10}
    initialScrollIndex={NaN}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</ForwardRef(SafeAreaView)>
`;
